daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -448,8 +447,12 @@ object Speedy {
     def execute(v: SValue, machine: Machine) = {
       v match {
         case SPAP(prim, args, arity) =>
-          val args2 = args.clone.asInstanceOf[util.ArrayList[SValue]]
-          val missing = arity - args2.size
+          val missing = arity - args.size
+          val newArgsLimit = Math.min(missing, newArgs.length)
+
+          // Keep some space free, because both `KFun` and `KPushTo` will add to the list.
+          val functionArgs = new util.ArrayList[SValue](args.size + newArgsLimit)
```

Comment:
Wonder what to call this. The context here is that we're applying `newArgs` to a partially applied primitive that already has `args` applied. `functionArgs` is a bit funny name here as both `args` and `newArgs` are also function arguments. Why not just use `args2` as it's `args` extended with `newArgs`. Or maybe call it `extendedArgs`?

Reply:
I'm happy to rename it to `extendedArgs`. 🙂